src/main/java/sirius/biz/storage/layer2/BasicBlobStorageSpace.java

```diff
@@ -263,10 +263,19 @@
     protected static Cache<String, String> blobKeyToPhysicalCache =
             CacheManager.createCoherentCache("storage-physical-keys");
 
+    protected static final String REMOVE_BY_FILENAME = "filename_remover";
+
     /**
      * Caches the {@link Blob blobs} that belong to certain paths
      */
-    protected static Cache<String, Blob> blobByPathCache = CacheManager.createCoherentCache("storage-paths");
+    protected static Cache<String, Blob> blobByPathCache = CacheManager.<Blob>createCoherentCache("storage-paths")
+                                                                       .addRemover(REMOVE_BY_FILENAME,
```

Comment:
**Optional:** Could in theory use `addValueBasedRemover` which directly passes the `entry.getValue()`